numpy/polynomial/legendre.py

```diff
@@ -1556,6 +1576,18 @@ def legfit(x, y, deg, rcond=None, full=False, w=None):
     c, resids, rank, s = la.lstsq(lhs.T/scl, rhs.T, rcond)
     c = (c.T/scl).T
 
+    # Expand c to include non-fitted coefficients, which are set to
+    # zero
+    if restricted_L:
```

Comment:
The insert function is not terribly efficient. It would be better to start with an array of zeros of appropriate shape and fill in the fitted coefficients.

I'm also not a fan of capitalization in variables if it can be avoided, which it usually can.